src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp

```diff
@@ -1879,109 +1887,26 @@ namespace OpenMS
     }
   }
 
-  bool TOPPViewBase::annotateMS1FromMassFingerprinting_(const FeatureMap& identifications)
+  void TOPPViewBase::annotateWithAMS()
   {
     LayerData& layer = getActiveCanvas()->getCurrentLayer();
-    if (layer.type == LayerData::DT_PEAK)
+    LayerAnnotatorAMS annotator;
+    if (!annotator.annotate(layer, log_, current_path_))
     {
-      IDMapper im;
-      Param p = im.getParameters();
-      p.setValue("rt_tolerance", 30.0);
-      im.setParameters(p);
-      log_->appendNewHeader(LogWindow::LogState::NOTICE, "Note", "Mapping matches with 30 sec tolerance and no m/z limit to spectra...");
-      im.annotate((*layer.getPeakDataMuteable()), identifications, true, true);
-      return true;
+      return;
```

Comment:
should a warning be written to the log?

Reply:
this is done internally by passing `log_`